geniza/footnotes/metadata_export.py

```diff
@@ -56,3 +56,39 @@ def get_export_data_dict(self, source):
             # construct directly to avoid extra db calls
             "admin_url": f"{self.url_scheme}{self.site_domain}/admin/footnotes/source/{source.id}/change/",
         }
+
+
+class FootnoteExporter(Exporter):
+    """
+    A subclass of :class:`geniza.common.metadata_export.Exporter` that
+    exports information relating to :class:`~geniza.footnotes.models.Footnote`.
+    """
+
+    model = Footnote
+    csv_fields = [
+        "document",  # ~ content object
+        "document_id",
+        "source",
+        "location",
+        "doc_relation",
+        "notes",
+        "url",
+        "content",
+        "admin_url",
+    ]
+
+    def get_queryset(self):
+        return self.queryset or self.model.objects.all().metadata_prefetch()
```

Comment:
So we're only running .metadata_prefetch in the event we don't have an incoming queryset (i.e. in the event of an admin interface select X to download operation). But in that event, are we running FootnoteAdmin.get_queryset which ends up running .metadata_prefetch? Still a little confused when we are and aren't using it. SourceExporter was using a custom queryset as manager?

Reply:
yeah, my assumption is that `metadata_prefetch` has already been called if we're given a queryset — because the only place right now where I know we call it with a subset of records is from the admin, and we're using the same prefetching there, so it should be applied already.

We already had a custom queryset for footnotes with other functions, so I just moved the existing admin fetching into a reusable method.